src/hotspot/share/opto/node.hpp

```diff
@@ -1128,6 +1129,21 @@ class Node {
   // Set control or add control as precedence edge
   void ensure_control_or_add_prec(Node* c);
 
+  // Visit all uses of the node and apply callback, bypassing nodes
+  // according to bypass. Note: definition appears after complete type
+  // definition of Unique_Node_List
+  template <typename Callback, typename Bypass>
+  void visit_uses(Callback callback, Bypass bypass);
```

Comment:
You should say in the comment that only "boundary" nodes get a callback.

Reply:
`bypass` could be misunderstood as "ignoring and not recursing".
Maybe you should say that you look through these nodes recursively, until you find a "boundary" node.
Maybe you could even swap the logic to "is_boundary_node"?

Reply:
Sure, I'll fix that.